src/com/hms_networks/americas/sc/taginfo/TagInfoManager.java

```diff
@@ -282,22 +301,194 @@ private static synchronized void processTagListEBDLine(String line) {
+   * stores the list. This must be done once at the start of the application and is handled when the
+   * TagInfoList is created.
+   *
+   * @throws IOException if int to string enumeration file read fails
+   * @throws JSONException if int to string enumeration JSON parse fails
+   */
+  private static void getIntStringEnumTags() throws IOException, JSONException {
+    if (enableIntToStringEnums) {
+      ArrayList tagList = new ArrayList();
+      JSONObject allTagIntToStringMappingsJson;
+      JSONArray tagIntToStringMappingsJsonArray;
+      allTagIntToStringMappingsJson =
+          new JSONObject(FileAccessManager.readFileToString(INT_TO_STRING_ENUMERATION_FILE_NAME));
+      tagIntToStringMappingsJsonArray =
+          (JSONArray) allTagIntToStringMappingsJson.get("enumeratedTagList");
+      final int tagListJsonArrayLength = tagIntToStringMappingsJsonArray.length();
+      for (int i = 0; i < tagListJsonArrayLength; i++) {
+        String tagValueAsString = tagIntToStringMappingsJsonArray.getString(i);
+        tagList.add(tagValueAsString);
+      }
+      intToStringEnumerationTagList = tagList;
+    }
+  }
+
+  /**
+   * For a single tag, get the array of string mappings corresponding to the given integer tag.
+   *
+   * @param tagName the name of the Ewon tag
+   * @return an array of string mappings corresponding to the current integer tag
+   * @throws IOException if int to string enumeration file read fails
+   * @throws JSONException if int to string enumeration JSON parse fails
+   */
+  private static String[] getIntStringEnumTagMappings(String tagName)
+      throws IOException, JSONException {
+    String[] tagStringMappings = null;
+    JSONObject tagIntToStringMappingsFileJson;
+    JSONObject tagIntToStringMappingsJson;
+    JSONArray tagIntToStringMappingsJsonArray;
+    if (enableIntToStringEnums) {
+      tagIntToStringMappingsFileJson =
+          new JSONObject(FileAccessManager.readFileToString(INT_TO_STRING_ENUMERATION_FILE_NAME));
+      tagIntToStringMappingsJson = (JSONObject) tagIntToStringMappingsFileJson.get("tags");
+      tagIntToStringMappingsJsonArray = (JSONArray) tagIntToStringMappingsJson.get(tagName);
+      final int length = tagIntToStringMappingsJsonArray.length();
+      int biggestIndex = 0;
+
+      /* Int to string enumerations are held in a JSON array of one key/value pair for each index in the array.
+       * To get the key of a key value pair, get the first(only) item returned by JSONObject.names()
+       * for the JSON object index.
+       */
+      final int tagIntToStringEnumStringKeyIndex = 0;
+      for (int jsonArrayIndex = 0; jsonArrayIndex < length; jsonArrayIndex++) {
+        String currentKey =
+            (String)
+                ((JSONObject) tagIntToStringMappingsJsonArray.get(jsonArrayIndex))
+                    .names()
+                    .get(tagIntToStringEnumStringKeyIndex);
+        int current = Integer.parseInt(currentKey);
```

Comment:
isnt there a NumberFormatException that can happen here?

Reply:
true, I added this to the javadocs for both this and the calling function